### PR TITLE
Fix for salvage staying indefinitely if magnet is sent away on cargo shuttle

### PR DESCRIPTION
**Pull request by dmnct**
## About the PR 
If the salvage magnet becomes paused (such as when it's on a cargo shuttle that gets recalled), now it will immediately lose the current salvage debris and enter the cooldown phase. 

Cooldown time is extended to also cover the time the salvage would have stayed originally, to prevent people from rapidly cycling through debris maps. If the magnet was still Attaching and there is no debris, the cooldown time is only extended by the remaining portion of the Attach phase.

Due to cooldowns now potentially lasting 6 minutes, which does NOT progress while the shuttle is away, specific cooldown time should be visible to players so they don't think the magnet got stuck on cooldown. #13214

**Media**

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Errant
- fix: Due to budget cuts, salvage magnets can no longer hold debris across interstellar distances. Expect potentially long cooldowns when interrupting an active salvage magnet by sending it away on a shuttle.

### CODE REVIEW

**Comment by dmnct** [comment]
> Pausing should not be wiping state, just have it detach on map changes.

I have created a new Event when specified items are on the shuttle as it departs. Currently it only watches for salvage magnets. I wanted it to watch an expandable list of components, since certain other items like the nuclear authentication disk, being recalled on the shuttle might be a point of interest, but I could not make it work with a list input

File: Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs [hunk 1]
```diff
@@ -356,6 +371,7 @@ private void SendToCargoMap(EntityUid uid, CargoShuttleComponent? component = nu
 
         UpdateShuttleCargoConsoles(component);
         _sawmill.Info($"Stashed cargo shuttle {ToPrettyString(uid)}");
+        ReportDepartingItems(component);
     }
```
Comment (metalgearsloth) [inline Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:375]: This doesn't fix the issue because it only handles 1 of many different map changes.

Comment (metalgearsloth) [reply Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:375]: I think the best thing to do for now is just iterate active magnets and check that the target is on the same map.

Comment (dmnct) [reply Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:375]: > I think the best thing to do for now is just iterate active magnets and check that the target is on the same map.

Is it okay to put the check in Update, running every frame?

Comment (metalgearsloth) [reply Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:375]: Yeah, there's only 1 magnet most of the time and I think the other alternatives might be insufficient (e.g. subbing to grid map changes or re-adding the map change event back).

**Comment by metalgearsloth** [comment]
@dmnct are you coming back to this?

**Comment by dmnct** [comment]
> @dmnct are you coming back to this?

I intend to work on it this week

**Comment by dmnct** [comment]
I gave it a shot but enthusiasm did not materialise. At least for now, the PR is dead

**Comment by AJCM-git** [comment]
> I gave it a shot but enthusiasm did not materialise. At least for now, the PR is dead. Should I close it?

You can re-open once you have more motivation

**Pull request closed without merging**